### PR TITLE
doc: fixing docstrings and API docs for gradient

### PR DESCRIPTION
**Pull request by davidsbatista**

### CODE REVIEW

File: integrations/gradient/src/haystack_integrations/components/embedders/gradient/gradient_document_embedder.py [hunk 1]
```diff
@@ -125,6 +135,7 @@ def _generate_embeddings(self, documents: List[Document], batch_size: int) -> Li
     def run(self, documents: List[Document]):
         """
         Embed a list of Documents.
+
         The embedding of each Document is stored in the `embedding` field of the Document.
 
         :param documents: A list of Documents to embed.
```
Comment (vblagoje) [inline integrations/gradient/src/haystack_integrations/components/embedders/gradient/gradient_document_embedder.py:141]: missing that returns with documents for example see this [one](https://github.com/deepset-ai/haystack/blob/main/haystack/components/connectors/openapi_service.py#L91)

Comment (davidsbatista) [reply integrations/gradient/src/haystack_integrations/components/embedders/gradient/gradient_document_embedder.py:141]: 👍🏽

File: integrations/gradient/src/haystack_integrations/components/embedders/gradient/gradient_document_embedder.py [hunk 2]
```diff
@@ -23,8 +23,10 @@ def _alt_progress_bar(x: Any) -> Any:
 class GradientDocumentEmbedder:
     """
     A component for computing Document embeddings using Gradient AI API.
+
     The embedding of each Document is stored in the `embedding` field of the Document.
 
+    Usage example:
```
Comment (vblagoje) [inline integrations/gradient/src/haystack_integrations/components/embedders/gradient/gradient_document_embedder.py:29]: Let's put data param name below in the pipeline run call?

File: integrations/gradient/src/haystack_integrations/components/embedders/gradient/gradient_document_embedder.py [hunk 3]
```diff
@@ -125,9 +135,14 @@ def _generate_embeddings(self, documents: List[Document], batch_size: int) -> Li
     def run(self, documents: List[Document]):
         """
         Embed a list of Documents.
+
         The embedding of each Document is stored in the `embedding` field of the Document.
 
         :param documents: A list of Documents to embed.
+        :returns:
+            A dictionary with the following keys:
+            - documents: The embedded Documents.
```
Comment (vblagoje) [inline integrations/gradient/src/haystack_integrations/components/embedders/gradient/gradient_document_embedder.py:144]: nit `documents` 🙏

File: integrations/gradient/src/haystack_integrations/components/embedders/gradient/gradient_text_embedder.py [hunk 4]
```diff
@@ -8,8 +8,9 @@
 @component
 class GradientTextEmbedder:
     """
-    A component for embedding strings using models hosted on Gradient AI (https://gradient.ai).
+    A component for embedding strings using models hosted on [Gradient AI](https://gradient.ai).
 
+    Usage example:
```
Comment (vblagoje) [inline integrations/gradient/src/haystack_integrations/components/embedders/gradient/gradient_text_embedder.py:13]: Something's off with the pipeline.run invocation! This example won't work.

File: integrations/gradient/src/haystack_integrations/components/generators/gradient/base.py [hunk 5]
```diff
@@ -16,6 +16,7 @@ class GradientGenerator:
     Queries the LLM using Gradient AI's SDK ('gradientai' package).
     See [Gradient AI API](https://docs.gradient.ai/docs/sdk-quickstart) for more details.
 
+    Usage example:
     ```python
     llm = GradientGenerator(base_model_slug="llama2-7b-chat")
```
Comment (vblagoje) [inline integrations/gradient/src/haystack_integrations/components/generators/gradient/base.py:21]: Let's add imports here

File: integrations/gradient/src/haystack_integrations/components/embedders/gradient/gradient_text_embedder.py (hunk 4 above)
Comment (vblagoje) [reply integrations/gradient/src/haystack_integrations/components/embedders/gradient/gradient_text_embedder.py:13]: Let's add imports as well

File: integrations/gradient/src/haystack_integrations/components/embedders/gradient/gradient_document_embedder.py (hunk 2 above)
Comment (vblagoje) [reply integrations/gradient/src/haystack_integrations/components/embedders/gradient/gradient_document_embedder.py:29]: Let's add imports

**Review by vblagoje — CHANGES_REQUESTED** [review]
Left a few comments. Someone has not coded these examples well - it seems to me

**Review by vblagoje — APPROVED** [review]
This now looks light years better than before. Great stuff @davidsbatista

**Pull request merged**